olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -832,9 +833,17 @@ DefaultCache::StorageOpenResult DefaultCacheImpl::SetupProtectedCache() {
   StorageSettings protected_storage_settings;
   protected_storage_settings.max_file_size = 32 * 1024 * 1024;
 
-  auto status = protected_cache_->Open(
-      settings_.disk_path_protected.get(), settings_.disk_path_protected.get(),
-      protected_storage_settings, OpenOptions::ReadOnly);
+  OpenOptions open_mode = settings_.openOptions;
```

Comment:
Please add a proper comment why we are doing this so that we know in the future it was not just a typo. Explain please that we are doing this for the sake of recovering from not compacted close and avoid high RAM usage on uncompacted databases with r/o environment.